Source/Core/Core/HW/GCMemcard/GCMemcardDirectory.cpp

```diff
@@ -39,6 +39,28 @@
 
 static const char* MC_HDR = "MC_SYSTEM_AREA";
 
+static std::string GenerateDefaultGCIFilename(const Memcard::DEntry& entry,
+                                              bool card_encoding_is_shift_jis)
+{
+  const auto string_decoder = card_encoding_is_shift_jis ? SHIFTJISToUTF8 : CP1252ToUTF8;
+  const auto strip_null = [](const std::string& s) {
```

Comment:
I think this could take a `string_view` and return a `string_view` to avoid constructing a string that is passed to FMT?